src/netex_idf/offers.rs

```diff
@@ -267,13 +283,105 @@ fn enhance_with_object_code(
     enhanced_routes
 }
 
+fn arrival_departure_times(el: &Element) -> Result<(Time, Time)> {
+    fn time(el: &Element, node_name: &str) -> Result<Time> {
+        Ok(el.try_only_child(node_name)?.text().parse()?)
+    }
+    let offset: u32 = match el.try_only_child("DepartureDayOffset")?.text().parse() {
+        Ok(offset) if offset > 0 => offset,
+        Ok(offset) if offset == 0 => 0,
```

Comment:
Why separating `>` and `==`. You should be able to just do
```rust
Ok(offset) if offset >= 0 => offset,
_ => 0
```
And since you're converting to a `u32` which cannot be negative, you can probably do just
```rust
let offset: u32 = el
    .try_only_child("DepartureDayOffset")?
    .text()
    .parse()
    .unwrap_or(0);
```

Reply:
Yep, I tried it too and `clippy` is actually complaining about matching with a condition `>= 0` which basically means `positive number`... and of course `u32` is positive. But it took me a minute to understand why `clippy` was complaining :smile: